Snowflake.Data.Tests/UnitTests/ConnectionPoolManagerTest.cs

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System.Collections.Generic;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Snowflake.Data.Core;
+using Snowflake.Data.Core.Session;
+using Moq;
+using Snowflake.Data.Client;
+using Snowflake.Data.Tests.Util;
+
+namespace Snowflake.Data.Tests.UnitTests
+{
+    [TestFixture, NonParallelizable]
+    class ConnectionPoolManagerTest
+    {
+        private readonly ConnectionPoolManager _connectionPoolManager = new ConnectionPoolManager();
+        private readonly string _connectionString1 = "database=D1;warehouse=W1;account=A1;user=U1;password=P1;role=R1;";
+        private readonly string _connectionString2 = "database=D2;warehouse=W2;account=A2;user=U2;password=P2;role=R2;";
+        private readonly SecureString _password = new SecureString();
+        private static PoolConfig s_poolConfig;
+
+        [OneTimeSetUp] 
+        public static void BeforeAllTests()
+        {
+            s_poolConfig = new PoolConfig();
+            SnowflakeDbConnectionPool.SetConnectionPoolVersion(ConnectionPoolType.MultipleConnectionPool);
+            SessionPool.SessionFactory = new MockSessionFactory();
+        }
+        
+        [OneTimeTearDown]
+        public void AfterAllTests()
+        {
+            s_poolConfig.Reset();
+            SessionPool.SessionFactory = new SessionFactory();
```

Comment:
SessionFactory is internal so should be s_.
I think we should rather have a method to change SessionPool attributes.

Reply:
I'm fixing naming convention.